docs/specs/oidc-sso.md

```diff
@@ -0,0 +1,176 @@
+
+## Implementation details
+
+### Server-side
+
+- Authorization Endpoint
+    - New query parameter `x_sso_enabled`
+        - When `x_sso_enabled=false`
+            - Equal to `x_suppress_idp_session_cookie=true`.
+            - The existing IdP session in cookie is ignored, and login will NOT create IdP Session.
+            - Mark refresh token's `sso_enabled` to `false`.
+        - When `x_sso_enabled=true`
+            - Equal to `x_suppress_idp_session_cookie=false`.
+            - Mark refresh token's `sso_enabled` to `true`.
+            - Refresh tokens and IdP sessions lifetime are managed separately (Remains unchanged). The only difference is the refresh token's lifetime will be shorter than or equal to the IdP session. e.g. If the IdP session will expire in 1 day, the refresh token will only have 1 day lifetime when it is issued.
+            - When updating the refresh token's last access time, its IdP session's last access time will also be updated. So that the IdP session won't be invalidated due to inactivity.
+    - `x_suppress_idp_session_cookie` remains unchanged for backward compatibility, but the new SDKs won't send this parameter anymore. It is replaced by `x_sso_enabled`.
+    - When the developer uses `x_sso_enabled=true` + `prompt=login`, the enduser will need to log in again. The new refresh token will be linked to the new IdP session, and the old IdP session will be overwritten and revoked.
+
+- Preserve `client_id` and `redirect_uri` query during the whole login flow.
+    - After authentication, redirect the enduser based on following whitelisted
+        1. Redirect URL in query param (must be whitelisted) 
+        2. Default redirect URL of the client
+        3. Post-login URL
+        4. `/settings`
+    - Default redirect URL of the client
+        - In the config, the first item of **redirect_uris** is the default implicitly
+        - Update the Portal UI to allow setting the default
+
+- Logout
+    - Revoke IdP session
+        - Check if any associate refresh tokens that have `sso_enabled=true`, revoke them
+    - Revoke refresh token
+        - If the refresh token's `sso_enabled` is `true`, invalidates its IdP session and all its siblings with `sso_enabled=true`
+        - If the refresh token's `sso_enabled` is `false` (default), revoke the refresh token only
+
+- "Continue as" screen
+    - When the enduser chooses **Login with another account**, the existing IdP session (including its refresh token with `sso_enabled=true`) will be revoked after the new authentication is completed.
+    - Add help text: `After you have signed in another account, your current session as user@example.com will terminate`.
+
+- Update application config
+    - For `x_application_type=spa`, change `post_logout_uris` to optional. Change the label in the Portal to **Post Logout Redirect URI (Legacy)**. See [Web SDK](#web-sdk) for details.
+
+- Session listing
```

Comment:
- SSO enabled: 1 entry per device
- Without SSO: 1 entry for each app on each device
- Session created by biometric will be handled separately.

Reply:
I suggest this.

Without SSO, we group refresh tokens for each app on each device and show the group as 1 entry. The IDP session is not grouped. For existing apps, the IDP session will be shown as 1 entry. For new apps, we do not create IDP session so it is not an issue.

This means visually we do not have group in the session list.

Reply:
> Without SSO, we group refresh tokens for each app on each device and show the group as 1 entry.

But can we match device reliably? If not, I think if an app only calls login without logout will have multiple entry, is fine...? (Most apps won't keep logout users right?)

Reply:
Conclusion: SSO disabled refresh token will not be grouped.